applicationinsights.ts

```diff
@@ -4,285 +4,261 @@ import AutoCollectExceptions = require("./AutoCollection/Exceptions");
         }
 
-        return ApplicationInsights;
+        return Configuration;
     }
 
     /**
      * Sets the state of automatic dependency correlation (enabled by default)
      * @param value if true dependencies will be correlated with requests
-     * @returns {ApplicationInsights} this class
+     * @returns {Configuration} this class
      */
     public static setAutoDependencyCorrelation(value: boolean) {
-        ApplicationInsights._isCorrelating = value;
-        if (ApplicationInsights._isStarted) {
-            ApplicationInsights._serverRequests.useAutoCorrelation(value);
+        _isCorrelating = value;
+        if (_isStarted) {
+            _serverRequests.useAutoCorrelation(value);
         }
 
-        return ApplicationInsights;
+        return Configuration;
     }
 
-     /**
-     * Enable or disable offline mode to cache events when client is offline (disabled by default)
+    /**
+     * Enable or disable disk-backed retry caching to cache events when client is offline (enabled by default)
+     * Note that this method only applies to the default client. Disk-backed retry caching is disabled by default for additional clients.
+     * For enable for additional clients, use client.channel.setUseDiskRetryCaching(true).
      * @param value if true events that occured while client is offline will be cached on disk
      * @param resendInterval. The wait interval for resending cached events.
-     * @returns {ApplicationInsights} this class
+     * @returns {Configuration} this class
      */
-    public static setOfflineMode(value: boolean, resendInterval?: number) {
-        ApplicationInsights._isOfflineMode = value;
-        if (ApplicationInsights.client && ApplicationInsights.client.channel){
-            ApplicationInsights.client.channel.setOfflineMode(value, resendInterval);
+    public static setUseDiskRetryCaching(value: boolean, resendInterval?: number) {
+        _isDiskRetry = value;
+        if (defaultClient && defaultClient.channel){
+            defaultClient.channel.setUseDiskRetryCaching(value, resendInterval);
         }
 
-        return ApplicationInsights;
+        return Configuration;
     }
 
     /**
      * Enables debug and warning logging for AppInsights itself.
-     * @param enableWarningLogging also show warnings
-     * @returns {ApplicationInsights} this class
+     * @param enableDebugLogging if true, enables debug logging
+     * @param enableWarningLogging if true, enables warning logging
+     * @returns {Configuration} this class
      */
-    public static enableVerboseLogging(enableWarningLogging = true) {
-        Logging.enableDebug = true;
-        Logging.disableWarnings = !enableWarningLogging;
-        return ApplicationInsights;
+    public static setInternalLogging(enableDebugLogging = false, enableWarningLogging = true) {
+        Logging.enableDebug = enableDebugLogging;
+        Logging.disableWarnings = enableWarningLogging;
```

Comment:
Is this behavior change intentional (`disable = enable` vs. `disable = !enable`)? It's doing the opposite now and I do not see it in the bug list being addressed by this PR.

Reply:
Yes it's intentional as before this was strangely placed as separate enable and disable functions. This is very much inconsistent with every other toggle that is one method that takes an on/off boolean.

Reply:
Actually I see you mean on `disableWarnings` rather than the change to the method name. Looks like a bug! Good catch